plugins/chat/assets/javascripts/discourse/services/chat.js

```diff
@@ -291,6 +291,37 @@ export default class Chat extends Service {
         this.chatChannelsManager.publicMessageChannelsWithActivity;
       directChannels =
         this.chatChannelsManager.directMessageChannelsWithActivity;
+
+      // If the active channel has no unread messages, we need to manually insert it into
+      // the list, so we can find the next/previous unread channel.
+      if (!activeChannel.hasUnread) {
+        let allChannels;
+
+        if (activeChannel.isDirectMessageChannel) {
+          allChannels = this.chatChannelsManager.directMessageChannels;
+        } else {
+          allChannels = this.chatChannelsManager.publicMessageChannels;
+        }
```

Comment:
Small nit but I like the ternary approach to reduce the verbosity a little (completely optional though):

```js
const allChannels = activeChannel.isDirectMessageChannel
  ? this.chatChannelsManager.directMessageChannels
  : this.chatChannelsManager.publicMessageChannels;
```